### PR TITLE
Refs #59343 - Add dropup support to dropdown menus. Increase default dropdown height

### PR DESCRIPTION
**Pull request by aaronransley**

### CODE REVIEW

File: app/assets/javascripts/fae/form/fae_chosen.js [hunk 1]
```diff
@@ -21,6 +21,23 @@
...
       $this.chosen(settings);
+
+      // add handling for bottom of viewport / drop-up support
+      $this.on('chosen:showing_dropdown', function(event, params) {
+        var chosen_container = $(event.target ).next('.chosen-container');
+        var dropdown = chosen_container.find('.chosen-drop');
+        var results_container_max_height = parseInt(dropdown.find('.chosen-results').css('max-height'));
+        var results_container_top_offset = 40; // to account for possibility of chosen search input
+        var dropdown_top = dropdown.offset().top - $(window).scrollTop();
+        var viewport_height = $(window).height();
```
Comment (tshedor) [inline app/assets/javascripts/fae/form/fae_chosen.js]: This is already available as `FCH.dimensions.wh`

File: app/assets/javascripts/fae/form/fae_chosen.js [hunk 2]
```diff
@@ -21,6 +21,23 @@
...
 
       $this.chosen(settings);
+
+      // add handling for bottom of viewport / drop-up support
+      $this.on('chosen:showing_dropdown', function(event, params) {
+        var chosen_container = $(event.target ).next('.chosen-container');
+        var dropdown = chosen_container.find('.chosen-drop');
+        var results_container_max_height = parseInt(dropdown.find('.chosen-results').css('max-height'));
+        var results_container_top_offset = 40; // to account for possibility of chosen search input
+        var dropdown_top = dropdown.offset().top - $(window).scrollTop();
```
Comment (tshedor) [inline app/assets/javascripts/fae/form/fae_chosen.js]: Use `FCH.$window` instead

File: app/assets/javascripts/fae/form/fae_chosen.js [hunk 3]
```diff
@@ -21,6 +21,23 @@
       }
 
       $this.chosen(settings);
+
+      // add handling for bottom of viewport / drop-up support
+      $this.on('chosen:showing_dropdown', function(event, params) {
+        var chosen_container = $(event.target ).next('.chosen-container');
```
Comment (tshedor) [inline app/assets/javascripts/fae/form/fae_chosen.js]: All jQuery variables should be preceded by a $, i.e. `var $chosen_container`

Comment (aaronransley) [inline app/assets/javascripts/fae/form/fae_chosen.js]: `$` was not in place for surrounding jQuery variables, so I avoided introducing it. Will refactor to include

**Pull request merged**